scripts/wrappers/common/utils.py

```diff
@@ -211,3 +225,64 @@ def set_service_expected_to_start(service, start=True):
     else:
         fd = os.open(lock, os.O_CREAT, mode=0o700)
         os.close(fd)
+
+
+def xable(action: str, addons: list, xabled_addons: list):
+    """Enables or disables the given addons.
+
+    Collated into a single function since the logic is identical other than
+    the script names.
+    """
+    actions = Path(__file__).absolute().parent / "../../../microk8s-resources/actions"
+    existing_addons = {sh.with_suffix('').name[7:] for sh in actions.glob('enable.*.sh')}
+
+    # Backwards compatibility with enabling multiple addons at once, e.g.
+    # `microk8s.enable foo bar:"baz"`
+    if all(a.split(':')[0] in existing_addons for a in addons) and len(addons) > 1:
+        for addon in addons:
+            if addon in xabled_addons and addon != 'kubeflow':
+                click.echo("Addon %s is already %sd." % (addon, action))
+            else:
+                addon, *args = addon.split(':')
+                subprocess.run([str(actions / ('%s.%s.sh' % (action, addon)))] + args)
+
+    # The new way of xabling addons, that allows for unix-style argument passing,
+    # such as `microk8s.enable foo --bar`.
+    else:
+        addon, *args = addons[0].split(':')
+
+        if addon in xabled_addons and addon != 'kubeflow':
+            click.echo("Addon %s is already %sd." % (addon, action))
+            sys.exit(0)
+
+        if addon not in existing_addons:
+            click.secho("Addon `%s` not found." % addon, fg='red', err=True)
+            click.echo("The available addons are:\n - %s" % '\n - '.join(existing_addons), err=True)
+            sys.exit(1)
+
+        if args and addons[1:]:
+            click.secho(
+                click.style(
+                    "Can't pass string arguments and flag arguments simultaneously!\n", fg='red'
```

Comment:
I like what you are doing here, error messages in red. Could we do this across all microk8s in a separate PR? If we do this coloring only here it will look odd.